backend/src/db/services/lastwills.service.ts

```diff
@@ -0,0 +1,78 @@
+import { InjectModel } from '@m8a/nestjs-typegoose'
+import {
+  Injectable,
+  Logger,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common'
+import { ReturnModelType } from '@typegoose/typegoose'
+import { ObjectId } from 'mongoose'
+import { CreateLastWillDto } from '../../last-wills/dto/create-lastwill.dto'
+import { UpdateLastWillDto } from '../../last-wills/dto/update-lastwill.dto'
+import { paymentPlans } from '../../payments/interfaces/payments'
+import { LastWill } from '../entities/lastwill.entity'
+import { UserService } from './user.service'
+
+@Injectable()
+export class LastWillsService {
+  private readonly logger = new Logger(LastWillsService.name)
+  constructor(
+    @InjectModel(LastWill)
+    private readonly lastWillModel: ReturnModelType<typeof LastWill>,
+    private readonly userService: UserService,
+  ) {}
+
+  async createOne(createLastWillDto: CreateLastWillDto, userId: ObjectId) {
+    const lastWillCount = await this.lastWillModel.countDocuments({
+      accountId: userId,
+    })
+    const user = await this.userService.findOneById(userId)
+    if (!user) throw new UnauthorizedException('Only Existing users can create')
+    const plan = user.paymentPlan
+    const allowedWills = Math.abs(paymentPlans[plan])
+    if (lastWillCount >= allowedWills)
+      throw new UnauthorizedException(
```

Comment:
Unauthorised is the wrong exception
From the mozilla docs:

> The HyperText Transfer Protocol (HTTP) 401 Unauthorized response status code indicates that the client request has not been completed because it lacks valid authentication credentials for the requested resource.

Prefer using 403 forbidden or 402 payment required. See also the [mozilla docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403)